numpyro/handlers.py

```diff
@@ -242,6 +243,66 @@ def process_message(self, msg):
             msg['stop'] = True
 
 
+class collapse(trace):
+    """
+    EXPERIMENTAL Collapses all sites in the context by lazily sampling and
+    attempting to use conjugacy relations. If no conjugacy is known this will
+    fail. Code using the results of sample sites must be written to accept
+    Funsors rather than Tensors. This requires ``funsor`` to be installed.
+    """
+    def __init__(self, *args, **kwargs):
```

Comment:
Ditto: update `__init__`, `__enter__` and `__exit__` as in https://github.com/pyro-ppl/pyro/pull/2641